### ISSUE TITLE
[spaced-comment] require two (N) spaces before a line comment?

### ISSUE THREAD

**Issue by quezak** [eslint/eslint#12907]
**What rule do you want to change?**
`spaced-comment`

**Does this change cause the rule to produce more or fewer warnings?**
the same

**How will the change be implemented? (New option, new default behavior, etc.)?**
Add an option to require two spaces before a line comment (`//`).
Some style guides (notably [Google's for C++](https://google.github.io/styleguide/cppguide.html#Implementation_Comments)) suggest this to have a better visual separation between the code and comments.

This could be a `beforeLineComment: number` option for the `spaced-comment` rule.

**Please provide some example code that this change will affect:**
```typescript
someFn();  // does stuff
// ^ OK with the option set to 2

someFn(); // does stuff
// ^ invalid with the option set to 2, OK if set to 1

someFn();// does stuff
// ^ invalid with the option set to 1 or 2, could be OK if set to 0

// does stuff
someFn();
// ^ option is ignored here, works only for line comments "beside" code
```

**What does the rule currently do for this code?**
Allow to configure the whitespace inside comment markers.

**What will the rule do after it's changed?**
Additionally allow to configure whitespace before a line comment besides code.

**Are you willing to submit a pull request to implement this change?**
Maybe, if it's not too complex -- I'm not familiar with eslint's internals for now.

**Comment by kaicataldo** [comment]
Adding a `minimum` option that can be specified as a number makes sense to me. One thing we'd have to work out is how this interacts with the `balanced` option. I think the logical thing would be for the minimum to apply to both the beginning and end of a block comment.

**Comment by kaicataldo** [comment]
I don't think is the responsibility of this rule, as it deals with whitespace after/within comment delimiters. This seems like a good case for a custom rule to me!

**Comment by quezak** [comment]
@kaicataldo yeah, when writing the issue I've also started to doubt if this should be an option to this existing rule, but wanted to see other opinions first -- thanks for replying! 

Can you briefly describe what's the course of action now? Should I wait for more evaluations, or actually try to write such a rule?

**Comment by kaicataldo** [comment]
Next steps would be to wait for consensus (whose definition is described [here](https://eslint.org/docs/maintainer-guide/issues#consensus), though I think it's unlikely this would be accepted, for the reasons outlined above.

If this is something you need now and you don't want to wait for the consensus process to complete (either accepting or rejecting the proposal), you're always free to make your own custom rules!
